### PR TITLE
Isentropic Example

### PR DESCRIPTION
**Pull request by tjwixtrom**
Adds example demonstrating usage of isentropic_interpolation on 4-dimensional data.

### CODE REVIEW

File: examples/Isentropic_Interpolation.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+# **Plotting the Isentropic Analysis**
+
+#####################
+# Set up our projection
+crs = ccrs.LambertConformal(central_longitude=-100.0, central_latitude=45.0)
+
+# Set up our array of latitude and longitude values and transform to
+# the desired projection.
+clons, clats = np.meshgrid(lon, lat)
+tlatlons = crs.transform_points(ccrs.PlateCarree(), clons, clats)
```
Comment (dopplershift) [inline examples/Isentropic_Interpolation.py]: Why transform manually? Instead you should be able to pass `transform=ccrs.PlateCarree()` to `contour` and friends.

File: examples/Isentropic_Interpolation.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+# components) will be linearly interpolated to isentropic space.
+
+isent_anal = metpy.calc.isentropic_interpolation(isentlevs, lev, tmp, relh, uwnd, vwnd, axis=1)
+
+####################
+# The output is a tuple, so now we will separate the variables to different names before
+# plotting.
+
+isentprs = isent_anal[0]
+isentrh = isent_anal[1]
```
Comment (dopplershift) [inline examples/Isentropic_Interpolation.py]: I get that doing this above will make the line ludicrously long, so how about here we use the more pythonic:
```python
isentprs, isentrh, isentu, isentv = isent_anal
```

**Comment by tjwixtrom** [comment]
Update reflects requested changes.

**Comment by tjwixtrom** [comment]
I have updated for current Siphon updates, let me know if this is the best way to build a query.

File: examples/Isentropic_Interpolation.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+###########################
+# Getting the data
+#
+# In this example, the latest GFS forecasts data from the National Centers for
+# Environmental Information (https://nomads.ncdc.noaa.gov) will be used, courtesy of the
+# Univeristy Corporation for Atmospheric Research Thredds Data Server.
+
+# Latest GFS Dataset
+cat = TDSCatalog('http://thredds.ucar.edu/thredds/catalog/grib/'
```
Comment (dopplershift) [inline examples/Isentropic_Interpolation.py]: Might be easier as:
```python
cat = TDSCatalog('http://thredds.ucar.edu/thredds/catalog/grib/'
                               'NCEP/GFS/Global_0p5deg/catalog.xml')
ncss = cat.latest.subset()
```
?

**Pull request merged**